wps_jobs/wps_jobs/weather_model_jobs/ecmwf_prediction_processor.py

```diff
@@ -0,0 +1,199 @@
+            self._process_prediction(prev_prediction, prediction, station, model_run, machine, False)
+
+            prev_prediction = prediction
+    
+    def _should_interpolate(self, prev_prediction: ModelRunPrediction, prediction: ModelRunPrediction) -> bool:
+        """Check if we should interpolate between the two predictions."""
+        prev_timestamp: datetime = prev_prediction.prediction_timestamp
+        next_timestamp: datetime = prediction.prediction_timestamp
+        assert next_timestamp > prev_timestamp, "Next timestamp must be greater than previous timestamp"
+        # Check if datetimes are on the same day
+        if prev_timestamp.date() == next_timestamp.date():
+            # Timestamps on the same day but surround 20:00 UTC should be interpolated
+            return prev_timestamp.hour <= 18 and next_timestamp.hour >= 21
+        
+        # Otherwise the next timestamp is on the next day and should always be interpolated for 20:00 UTC
+        return True
+
+    def _weather_station_prediction_initializer(self, station: WeatherStation, model_run: PredictionModelRunTimestamp, prediction: ModelRunPrediction) -> WeatherStationModelPrediction:
+        """Initialize a WeatherStationModelPrediction object."""
+        station_prediction = self.model_run_repository.get_weather_station_model_prediction(station.code, model_run.id, prediction.prediction_timestamp)
+        if station_prediction is None:
+            station_prediction = WeatherStationModelPrediction()
+        station_prediction = WeatherStationModelPrediction()
+        station_prediction.station_code = station.code
+        station_prediction.prediction_model_run_timestamp_id = model_run.id
+        station_prediction.prediction_timestamp = model_run.prediction_run_timestamp
+        return station_prediction
+    
+    
+    def _process_prediction(self, prev_prediction: ModelRunPrediction, prediction: ModelRunPrediction, station: WeatherStation, model_run: PredictionModelRunTimestamp, machine: StationMachineLearning, prediction_is_interpolated: bool):
+        """Create a WeatherStationModelPrediction from the ModelRunPrediction data."""
+        station_prediction = self._weather_station_prediction_initializer(station, model_run, prediction)
+        station_prediction.tmp_tgl_2 = prediction.get_temp()
+        station_prediction.tmp_tgl_2 = prediction.get_rh()
+        station_prediction.apcp_sfc_0 = float(0.0)
+
+        station_prediction.apcp_sfc_0 = prediction.get_precip()
+        station_prediction.precip_24h = self._calculate_past_24_hour_precip(station, model_run, prediction, station_prediction)
+        station_prediction.delta_precip = self._calculate_delta_precip(prev_prediction, station_prediction)
+
+        station_prediction.wind_tgl_10 = prediction.get_wind_speed()
+        station_prediction.wdir_tgl_10 = prediction.get_wind_direction()
+
+        if prediction_is_interpolated:
```

Comment:
The logic here seems off a little bit. When `prediction_is_interpolated == True` we've arrived in this function from this line in `_process_model_run_for_station`:
```
self._process_prediction(prev_prediction, noon_prediction, station, model_run, machine, True)
```.

So we have a noon_prediction but the code in this block is not interpolating between the prediction before noon and the noon prediction which we don't want to do. Can we not just use the noon_prediction for the station machine learning?

Reply:
Thanks for catching! This was a lift and shift from `common_model_fetchers` which does this:

```
        if prediction_is_interpolated:
            # Dealing with a numerical weather model that only has predictions at 3 hour intervals,
            # so no 20:00 UTC prediction available in the trained linear regression
            self._add_interpolated_bias_adjustments_to_prediction(station_prediction, machine)

        else:
            # No interpolation required
            self._add_bias_adjustments_to_prediction(station_prediction, machine)
```

I think where the error creeped in is that it originally used the `station_prediction`, I'm not sure if that's correct either..

Reply:
Ok this commit should remove that issue and simplify the code a bit. A noon interpolated prediction only has noon interpolated bias adjustments applied.